src/main/scala/com/mongodb/spark/connection/MongoClientRefCounter.scala

```diff
@@ -18,35 +18,27 @@
 //scalastyle:on
 package com.mongodb.spark.connection
 
-import scala.annotation.tailrec
-import scala.collection.concurrent.TrieMap
-
-import com.mongodb.MongoClient
+import java.util.concurrent.atomic.AtomicInteger
 
 /**
  * Atomically counts references to MongoClients
  */
 private class MongoClientRefCounter {
 
-  private val mongoClientCounts = new TrieMap[MongoClient, Int]
+  private val mongoClientCounts = new AtomicInteger(0)
```

Comment:
Not sure if this class is pulling enough weight to remain separate, now that it just wraps an AtomicInteger.  I also find the name confusing, since there is nothing MongoClient-specific about it anymore.